packages/mui-base/src/useDropdown/dropdownReducer.ts

```diff
@@ -9,7 +23,10 @@ export function dropdownReducer(state: DropdownState, action: DropdownAction): D
     case DropdownActionTypes.toggle:
       return { open: !state.open };
     case DropdownActionTypes.open:
-      return { open: true };
+      return {
+        open: true,
+        keyPressedToOpen: getKeyboardEventKey(action.event),
```

Comment:
To be consistent (and more generic), let's store the event that caused the `open` state to be changed:

```tsx
case DropdownActionTypes.blur:
  return { open: false, changeReason: action.event };
case DropdownActionTypes.escapeKeyDown:
  return { open: false, changeReason: action.event };
case DropdownActionTypes.toggle:
  return { open: !state.open, changeReason: action.event };
case DropdownActionTypes.open:
  return { open: true, changeReason: action.event };
case DropdownActionTypes.close:
  return { open: false, changeReason: action.event };
```

Also, add the `changeReason` field to the `DropdownState` type definition as non-optional.